src/vs/platform/opener/browser/link.ts

```diff
@@ -50,17 +52,24 @@ export class Link extends Disposable {
 			.event;
 		const onOpen = Event.any<EventLike>(onClick, onEnterPress);
 
+		this.handler = (href) => openerService.open(href, { allowCommands: true });
+		this._register({ dispose: () => this.handler = undefined! });
+
 		this._register(onOpen(e => {
 			EventHelper.stop(e, true);
 			if (!this.disabled) {
-				openerService.open(link.href, { allowCommands: true });
+				this.handler(link.href);
 			}
 		}));
 
 		this.disabled = false;
 		this.applyStyles();
 	}
 
+	setHandler(handler: (href: string) => void) {
+		this.handler = handler;
+	}
+
```

Comment:
Let's just add an `options?` bag to the ctor of `Link` instead of a mutable `setHandler`.

Reply:
Okay, I initially didn't just because the injected `openerService` means everywhere this is constructed would need to explicitly pass the empty bag or undefined.